[FreeBSD] Module thekla_unwrap doesn't compile due to wrong -DNV_OS_* preprocessor parameter

**Godot version:**
master (3.0beta2) bc513df custom-build

**OS/device including version:**
FreeBSD 11.0-RELEASE-p1 r306420
Intel(R) Core(TM) i3-2330M CPU @ 2.20GHz (2195.06-MHz K8-class CPU)

**Issue description:**
C++ compiler tries to compilate source code for Linux instead specific parts for FreeBSD, causing wrong functions being called.
```
In file included from thirdparty/thekla_atlas/nvcore/Memory.cpp:10:
/usr/include/malloc.h:3:2: error: "<malloc.h> has been replaced by <stdlib.h>"
#error "<malloc.h> has been replaced by <stdlib.h>"
 ^
thirdparty/thekla_atlas/nvcore/Memory.cpp:138:12: error: use of undeclared
      identifier 'memalign'
    return memalign(alignment, size);
           ^
2 errors generated.
scons: *** [thirdparty/thekla_atlas/nvcore/Memory.x11.tools.64.llvm.o] Error 1
In file included from thirdparty/thekla_atlas/nvcore/Debug.cpp:7:
thirdparty/thekla_atlas/nvcore/StdStream.h:164:26: error: use of undeclared
      identifier 'fwrite_unlocked'
            return (uint)fwrite_unlocked(data, 1, len, m_fp);
                         ^
thirdparty/thekla_atlas/nvcore/StdStream.h:214:26: error: use of undeclared
      identifier 'fread_unlocked'
            return (uint)fread_unlocked(data, 1, len, m_fp);

                         ^
2 errors generated.
```
**Steps to reproduce:**
```
git clone https://github.com/godotengine/godot.git
cd godot
scons platform=x11 use_llvm=yes -j 4
```

- [ ] I found and add info at [PullRequest#12190](https://github.com/godotengine/godot/pull/12190), and with the info received, I developed a possible solution proposed here.

**Proposed solution:**
Using platform/x11/detect.py as reference, I added platform.system() calls to obtain Linux, FreeBSD or OpenBSD strings, and use the thirdparty/thekla_atlas/nvcore/nvcore.h NV_OS_* definitions as needed. The implementation of Python for platform.system() is uname.system(), which according to Wikipedia, returns 'OpenBSD' in OpenBSD.

The diff is:
```
diff modules/thekla_atlas/SCsub
2a3
> import platform
63c64,71
<         env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_LINUX", "-DPOSH_COMPILER_GCC"])
---
>         if platform.system() == "FreeBSD":
>                 env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_FREEBSD", "-DPOSH_COMPILER_GCC"])
>         elif platform.system() == "OpenBSD":
>                 env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_OPENBSD", "-DPOSH_COMPILER_GCC"])
>         else:
>                 # if not specifically one of the *BSD, then use LINUX as default
>                 env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_LINUX", "-DPOSH_COMPILER_GCC"])
> 
```

--- a/modules/thekla_unwrap/SCsub
+++ b/modules/thekla_unwrap/SCsub
@@ -1,5 +1,7 @@
 #!/usr/bin/env python
 
+import platform
+
 Import('env')
 Import('env_modules')
 
@@ -60,7 +62,13 @@ if env['builtin_thekla_atlas']:
         env_thekla_unwrap.Append(CXXFLAGS="-std=c++11")
 
     if env["platform"] == 'x11':
-        env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_LINUX", "-DPOSH_COMPILER_GCC"])
+        # if not specifically one of the *BSD, then use LINUX as default
+        if platform.system() == "FreeBSD":
+            env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_FREEBSD", "-DPOSH_COMPILER_GCC"])
+        elif platform.system() == "OpenBSD":
+            env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_OPENBSD", "-DPOSH_COMPILER_GCC"])
+        else:
+            env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_LINUX", "-DPOSH_COMPILER_GCC"])
     elif env["platform"] == 'osx':
         env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_DARWIN", "-DPOSH_COMPILER_GCC"])
     elif env["platform"] == 'windows':
@@ -69,6 +77,6 @@ if env['builtin_thekla_atlas']:
         else:
             env_thekla_unwrap.Append(CCFLAGS=["-DNV_OS_MINGW", "-DNV_CC_GNUC", "-DPOSH_COMPILER_GCC", "-U__STRICT_ANSI__"])
             env.Append(LIBS=["dbghelp"])
-        
+
 # Godot source files
 env_thekla_unwrap.add_source_files(env.modules_sources, "*.cpp")

